src/models/devices.jl

```diff
@@ -1,9 +1,4 @@
-"""
-Add a service to a device.
-
-Throws ArgumentError if the service is already attached to the device.
-"""
-function add_service!(device::Device, service::Service)
+function add_service_internal!(device::Device, service::Service)
```

Comment:
@jd-lara Please challenge this change if you think it could be problematic.  I made it because I think the former implementation was a bit unsafe.  A user could add a service to a device without proper checks.  The public interface is `add_service!(sys, services, contributing_devices)`

Reply:
I think this is fine, it might break some scripts but not PSI code.